### ISSUE TITLE
userDataDir

### ISSUE THREAD

**Issue by mahdisky** [emre-gon/Selenium.WebDriver.UndetectedChromeDriver#13]
how to use this parameter userDataDir?

options.AddArgument("user-data-dir="+ userDataDir);

not work

**Comment by firecrauter** [comment]
```
ISet<string> driverarguments = new HashSet<string>();
            string thisdir= Environment.CurrentDirectory + @"\";
            string dirAllUsers = thisdir + "usersdata"+ @"\";
            string username_sanitize = "pepe";
            string full_path = dirAllUsers + username_sanitize;
            if (!Directory.Exists(full_path))
            {
                Directory.CreateDirectory(full_path);
            }
            driverarguments.Add("user-data-dir=" + full_path);
            Sl.Selenium.Extensions.Chrome.ChromeDriverParameters chromeDriverParameters = new()
            {
                Headless = false,
                DriverArguments = driverarguments,
                Timeout = new TimeSpan(0, 0, 22)
             
            };
Driver = UndetectedChromeDriver.Instance(chromeDriverParameters);
```

chrome://version/
![Sin título](https://user-images.githubusercontent.com/7044466/197399784-5ef19d04-e8e9-493f-9770-54c0b8b8f86c.png)

**Comment by RifHut** [comment]
> ```
> ISet<string> driverarguments = new HashSet<string>();
>             string thisdir= Environment.CurrentDirectory + @"\";
>             string dirAllUsers = thisdir + "usersdata"+ @"\";
>             string username_sanitize = "pepe";
>             string full_path = dirAllUsers + username_sanitize;
>             if (!Directory.Exists(full_path))
>             {
>                 Directory.CreateDirectory(full_path);
>             }
>             driverarguments.Add("user-data-dir=" + full_path);
>             Sl.Selenium.Extensions.Chrome.ChromeDriverParameters chromeDriverParameters = new()
>             {
>                 Headless = false,
>                 DriverArguments = driverarguments,
>                 Timeout = new TimeSpan(0, 0, 22)
>              
>             };
> Driver = UndetectedChromeDriver.Instance(chromeDriverParameters);
> ```
> 
> chrome://version/ ![Sin título](https://user-images.githubusercontent.com/7044466/197399784-5ef19d04-e8e9-493f-9770-54c0b8b8f86c.png)

@firecrauter i think u forgot something at Sl.Selenium.Extensions.Chrome.ChromeDriverParameters chromeDriverParameters = new() xxxxxxxxxxxxx;

**Comment by firecrauter** [comment]
> @RifHut 
>i think u forgot something at Sl.Selenium.Extensions.Chrome.ChromeDriverParameters chromeDriverParameters = new() xxxxxxxxxxxxx;

The code works for me, what do you think is missing?
(It was in the previous update, I don't know if anything has changed)

**Comment by RifHut** [comment]
> new()

missing new ChromeDriverParameters() !?

**Comment by firecrauter** [comment]
> > new()
> 
> missing new ChromeDriverParameters() !?

![image](https://user-images.githubusercontent.com/7044466/199777978-bbb62605-93b1-442c-af27-087c493eb5a0.png)
Need c# 9 or greater.

try this:
```
            Sl.Selenium.Extensions.Chrome.ChromeDriverParameters chromeDriverParameters = new();
            chromeDriverParameters.Timeout = new TimeSpan(0, 0, 22);
            chromeDriverParameters.DriverArguments = driverarguments;
            chromeDriverParameters.Headless = false;
```

**Comment by RifHut** [comment]
> Sl.Selenium.Extensions.Chrome.ChromeDriverParameters chromeDriverParameters = new();
>             chromeDriverParameters.Timeout = new TimeSpan(0, 0, 22);
>             chromeDriverParameters.DriverArguments = driverarguments;
>             chromeDriverParameters.Headless = false;

Yes u are right dude , I'm Working in Old Project with .Net framework v4.8 , that's why I get errors,
![image](https://user-images.githubusercontent.com/51222371/199846615-d23e3178-a0c6-4cb5-a33f-d5a41d66e981.png)

All Ok with framework v4.8 , but still detected on cloudflare , the example in this project "Tests" work fine. (net 6.0).

2 - also the shell window of undetect_chromedrive.exe still showing !

Thank You!

**Comment by firecrauter** [comment]
> Yes u are right dude , I'm Working in Old Project with .Net framework v4.8 , that's why I get errors
> 
> All Ok with framework v4.8 , but still detected on cloudflare , the example in this project "Tests" work fine. (net 6.0).
> 2 - also the shell window of undetect_chromedrive.exe still showing !
> 
> Thank You!

just in case, there is [openbulletv2](https://github.com/openbullet/OpenBullet2), which is being updated and uses net6

**Comment by RifHut** [comment]
> > Yes u are right dude , I'm Working in Old Project with .Net framework v4.8 , that's why I get errors
> > All Ok with framework v4.8 , but still detected on cloudflare , the example in this project "Tests" work fine. (net 6.0).
> > 2 - also the shell window of undetect_chromedrive.exe still showing !
> > Thank You!
> 
> just in case, there is [openbulletv2](https://github.com/openbullet/OpenBullet2), which is being updated and uses net6

You are right that's what i

> > Yes u are right dude , I'm Working in Old Project with .Net framework v4.8 , that's why I get errors
> > All Ok with framework v4.8 , but still detected on cloudflare , the example in this project "Tests" work fine. (net 6.0).
> > 2 - also the shell window of undetect_chromedrive.exe still showing !
> > Thank You!
> 
> just in case, there is [openbulletv2](https://github.com/openbullet/OpenBullet2), which is being updated and uses net6

I previously developed this SB project, but it has an old framework, the best I would do is moving to OB2, thanks dude.
i'm newbie at c# , i'm so sorry :) !
